src/main/java/com/faforever/client/preferences/LanguageInfo.java

```diff
@@ -0,0 +1,80 @@
+package com.faforever.client.preferences;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+
+public enum LanguageInfo {
+  AUTO("auto", null, "System", 0), EN("en", "UK", "English", 1), DE("de", "DE", "Deutsch", 2), RU("ru", "RU", "ру́сский", 3);
+
+  private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private final String languageCode;
+  private final String countryCode;
+  private final String dispayName;
+  private final int position;
+
+  LanguageInfo(String languageCode, String countryCode, String displayName, int position) {
+    this.countryCode = countryCode;
+    this.dispayName = displayName;
+    this.languageCode = languageCode;
+    this.position = position;
+  }
+  public static LanguageInfo getLanuageInfoByLanguageCode(String code) {
+    for (LanguageInfo LI : LanguageInfo.values()) {
+      if (LI.getLanguageCode().equals(code)) {
+        return LI;
+      }
+    }
+    logger.warn("No matching LanguageInfo found for LangaugeCode: " + code);
+    return null;
+  }
+  public static LanguageInfo getLanuageInfoByCountryCode(String code) {
+    for (LanguageInfo LI : LanguageInfo.values()) {
+      if (LI.getCountryCode().equals(code)) {
+        return LI;
+      }
+    }
+    logger.warn("No matching LanguageInfo found for CountryCode: " + code);
+    return null;
+  }
```

Comment:
Also, see how other enums in the project resolve from a value to an enum (using static maps) in order to reduce boilerplate and inefficient looping code.